synapse/replication/tcp/resource.py

```diff
@@ -104,18 +89,12 @@ def __init__(self, hs):
             self.federation_sender = hs.get_federation_sender()
 
         self.notifier.add_replication_callback(self.on_notifier_poke)
-        self.notifier.add_remote_server_up_callback(self.send_remote_server_up)
 
         # Keeps track of whether we are currently checking for updates
         self.is_looping = False
         self.pending_updates = False
 
-        hs.get_reactor().addSystemEventTrigger("before", "shutdown", self.on_shutdown)
-
-    def on_shutdown(self):
-        # close all connections on shutdown
-        for conn in self.connections:
-            conn.send_error("server shutting down")
```

Comment:
do we not still need this?

Reply:
Good Q. Its only used so that on the worker it gets logged, so its not a big deal if it goes. We can keep it easily enough, but it belongs in `protocol.py` (and probably shouldn't be server specific). Thoughts?

Reply:
I don't have strong feelings. bin it and bring it back later if we miss it?